test/framework/toy_build.py

```diff
@@ -862,9 +863,10 @@ def test_toy_hierarchical_subdir_user_modules(self):
         toy_mod = os.path.join(home, 'modules', 'all', openmpi_mod_subdir, 'toy', '0.0' + mod_ext)
         toy_modtxt = read_file(toy_mod)
 
-        for modname in ['FFTW', 'OpenBLAS', 'ScaLAPACK']:
-            regex = re.compile('load.*' + modname, re.M)
-            self.assertTrue(regex.search(toy_modtxt), "Pattern '%s' found in: %s" % (regex.pattern, toy_modtxt))
+        # No math libs in original toolchain, --try-toolchain is too clever to upgrade it beyond necessary
+        # for modname in ['FFTW', 'OpenBLAS', 'ScaLAPACK']:
+        #     regex = re.compile('load.*' + modname, re.M)
+        #     self.assertTrue(regex.search(toy_modtxt), "Pattern '%s' found in: %s" % (regex.pattern, toy_modtxt))
```

Comment:
I can remove them, the issue is there is no longer math libraries in the final output as `--try-toolchain` only upgrade them to `gompi` rather than `goolf`

Reply:
Then change the check to verify that the math libraries are *not* in the output anymore, and add a comment to explain why not?